src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTest.java

```diff
@@ -0,0 +1,86 @@
+package com.spanish_inquisition.battleship.server;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.List;
+
+import static com.spanish_inquisition.battleship.common.AppLogger.defaultLevel;
+import static com.spanish_inquisition.battleship.common.AppLogger.initializeLogger;
+import static com.spanish_inquisition.battleship.common.AppLogger.logger;
+
+public class BattleshipServerTest {
```

Comment:
I think that it's integration test. Name of the class could have "IT" at end of name.